### PR TITLE
cli: deprecate 'cockroach quit'

### LINKED ISSUE #47608: cli: deprecate `cockroach quit`
With the advent of `cockroach node drain`, better support for signal-based termination (eg Ctrl+C) and the move towards emphasizing orchestration and service managers, there is much less need for `cockroach quit` overall.

Also it has confusing semantics because it does not guarantee the server process has terminated when `quit` terminates.

so we're considering deprecating `quit` altogether.

cc @bdarnell @jseldess

### PR DESCRIPTION
**Pull request by knz**
tldr: `cockroach quit` marked deprecated in 20.2, will be removed in 21.1

Fixes  #47608

With the advent of `cockroach node drain`, better support for
signal-based termination (eg Ctrl+C) and the move towards emphasizing
orchestration and service managers, there is much less need for
`cockroach quit` overall.

Also it has confusing semantics because it does not guarantee the
server process has terminated when `quit` terminates.

Release note (cli change): The command `cockroach quit` is now
deprecated. Deployments are encouraged to use the new `cockroach node
drain` command instead, followed by process termination using
their standard service/process manager or orchestration tool.

### CODE REVIEW

**Review by bdarnell — COMMENTED** [review]
In general, I'd prefer to *stop recommending* a feature before we mark it as deprecated. That means starting with docs and removing all use of `cockroach quit` from guides and tutorials. 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/47758#-:-M5Xudd11RH2Lr9kwFIq:b-ot2a8y)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @bdarnell, @jseldess, and @knz)

___
*[pkg/cli/quit.go, line 38 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/47758#-M5XuwiICQwCrjPWw1su:-M5XuwiJCLSSjipRRktd:b-ktclo7) ([raw file](https://github.com/cockroachdb/cockroach/blob/2243a39f837bc2be2ee5d300a13d80d97d405b87/pkg/cli/quit.go#L38)):*
> ```Go
> 	Long: `
> This command is deprecated.
> Instead, use 'cockroach node drain' to stop all activity on a server
> ```

I think it makes too many assumptions to say that `node drain` is a substitute for `quit`. In most cases, shouldn't the orchestration layer be responsible for draining the node? I think direct use of `node drain` will be rare. I'd just say "Instead, use your process manager or orchestration layer to terminate the process, or send it a termination signal directly."

**Comment by knz** [comment]
Superseded by #49668
